sfall/LoadGameHook.cpp

```diff
@@ -276,6 +277,45 @@ static void __declspec(naked) LoadGame() {
 		retn;
 	}
 }
+
+void ReadExtraGameMsgFiles()
+{
+	int read;
+	std::string names;
+
+	names.resize(256);
+
+	while ((read = GetPrivateProfileStringA("Misc", "ExtraGameMsgFileList", "",
+		(LPSTR)names.data(), names.size(), ".\\ddraw.ini")) == names.size() - 1)
+		names.resize(names.size() + 256);
+
+	if (names.empty())
+		return;
+
+	names.resize(names.find_first_of('\0'));
+	names.append(",");
+
+	int begin = 0;
+	int end;
+	int length;
+
+	while ((end = names.find_first_of(',', begin)) != std::string::npos)
+	{
+		length = end - begin;
+
+		if (length > 0)
+		{
+			std::string path = "game\\" + names.substr(begin, length) + ".msg";
```

Comment:
My test didn't work. I created both foo.msg and bar.msg in game folder with {100}{}{Test message from foo/bar.} lines. Included these lines in artemple.ssl:
> display_msg("Message test started.");
> display_msg(message_str_game(0x2000, 100));
> display_msg(message_str_game(0x2001, 100));
> display_msg("Message test finished.");

When I start a new game, it only displays the first line "Message test started." in the message box. The other lines don't display. If I changed 0x2000/0x2001 to 0x1000/0x1001, all four lines display correctly.